docs: add component documentation steering file (#529)
db-ux-design-system/db-ux-design-system.github.io: .kiro/steering/component-documentation.md

2. **Create the Content Page** named `📋 Content [Component Name]`
3. **Write the structured text** following the schema above
4. **Create Guideline pages** for each Do/Dont pair
5. **Add visual examples** to each Guideline page (screenshots or component instances)
6. **Update the content text** with the correct node-id URLs pointing to the Guideline pages

### Adding a New Guideline to an Existing File

1. Create two new pages: `Guideline: [Topic] (do)` and `Guideline: [Topic] (dont)`
2. Add visual containers with example content to each page
3. Get the node-IDs of the new pages
4. Add a new `=== GUIDELINE:` block to the Content Page text with the correct URLs

### Content Writing Guidelines

| Aspect | Guidance |
|--------|----------|
| Description | Start with a clear one-sentence summary. Follow with 2–4 bullet points highlighting key features or use cases |
| Guideline Headline | Short, descriptive title (e.g. "ControlPanel Position", "Navigation Depth") |
| Guideline Text | Explain the rationale / when to apply this pattern. Keep concise (1–2 sentences) |
| Do Caption | Describe what the correct example shows – phrased as instruction ("Use X for Y") |

> | Dont Caption | Describe what's wrong – phrased as prohibition ("Do not use X when Y") |

Reviewer: ```suggestion
| Dont Caption | Describe what's wrong – phrased as prohibition ("Don't use X when Y") |
```

- **Line breaks matter** – each field (Headline, Text, Do, Dont) must be on its own line
- **URLs must be valid Figma URLs** – the platform resolves them to extract the visual content
- **Pipe separator** in Do/Dont and RELATED is ` | ` (space-pipe-space)

---

## Example: Complete Content Page Text

```
COMPONENT: Shell Desktop

DESCRIPTION:
The Shell Desktop component provides the outermost layout structure for desktop applications. It combines the ControlPanel with optional SubNavigation and a content area.
- Provide consistent page structure for desktop web applications
- Combine navigation, branding, and content areas in a single layout component
- Support different ControlPanel positions (Top, Left) and SubNavigation configurations

===

RESOURCES:
React:
Vue:
Angular:
Web C.:

===

GUIDELINE:
Headline: ControlPanel Position
Text: Choose the ControlPanel position based on navigation complexity. Top works for simple apps; Left suits applications with many navigation items or deep structures.
Do: https://www.figma.com/design/RcpbbqfJwjNRlOwPhAX1im/Doku--Shell?node-id=1-1132 | Use the Top position for applications with few top-level navigation entries.

> Dont: https://www.figma.com/design/RcpbbqfJwjNRlOwPhAX1im/Doku--Shell?node-id=1-1133 | Do not use the Left position when only 2–3 navigation items exist. The vertical panel takes up space without benefit.

Reviewer: ```suggestion
Dont: https://www.figma.com/design/RcpbbqfJwjNRlOwPhAX1im/Doku--Shell?node-id=1-1133 | Don't use the Left position when only 2–3 navigation items exist. The vertical panel takes up space without benefit.
```

===

GUIDELINE:
Headline: SubNavigation Placement
Text: Combine the ControlPanel and SubNavigation positions to support your information architecture. ControlPanel Top + SubNavigation Left works well for complex B2B applications.
Do: https://www.figma.com/design/RcpbbqfJwjNRlOwPhAX1im/Doku--Shell?node-id=3-1110 | Use SubNavigation Left for pages with many sub-level entries that benefit from a persistent vertical overview.

> Dont: https://www.figma.com/design/RcpbbqfJwjNRlOwPhAX1im/Doku--Shell?node-id=3-1112 | Do not switch SubNavigation position when navigating between top-level sections. Users expect the navigation layout to stay stable.

Reviewer: ```suggestion
Dont: https://www.figma.com/design/RcpbbqfJwjNRlOwPhAX1im/Doku--Shell?node-id=3-1112 | Don't switch SubNavigation position when navigating between top-level sections. Users expect the navigation layout to stay stable.
```

===

GUIDELINE:
Headline: Navigation Depth
Text: Multi-level navigation (Popover with nested levels) helps users find deeply nested content. Apply multi-level in either the ControlPanel or the SubNavigation – never in both.
Do: https://www.figma.com/design/RcpbbqfJwjNRlOwPhAX1im/Doku--Shell?node-id=3-1113 | Use multi-level navigation in one area only, and keep the other flat.

> Dont: https://www.figma.com/design/RcpbbqfJwjNRlOwPhAX1im/Doku--Shell?node-id=3-1114 | Do not use multi-level Popovers in both ControlPanel and SubNavigation at the same time. This makes orientation harder for users.

Reviewer: ```suggestion
Dont: https://www.figma.com/design/RcpbbqfJwjNRlOwPhAX1im/Doku--Shell?node-id=3-1114 | Don't use multi-level Popovers in both ControlPanel and SubNavigation at the same time. This makes orientation harder for users.
```
